src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7593,6 +7598,11 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                                 method,
                                 arguments[0],
                                 method.ReturnType);
+                            // Check for some required well-known members. They may not be needed
+                            // during lowering, but it's simpler to always require them to prevent
+                            // the user from getting surprising errors when optimizations fail
+                            _ = GetWellKnownTypeMember(Compilation, WellKnownMember.System_Range__get_Start, diagnostics, syntax: syntax);
+                            _ = GetWellKnownTypeMember(Compilation, WellKnownMember.System_Range__get_End, diagnostics, syntax: syntax);
```

Comment:
>GetWellKnownTypeMember [](start = 32, length = 22)

consider factoring the GetWellKnownTypeMember checks into a local function and calling here and above.